xchange-core/src/main/java/org/knowm/xchange/service/trade/TradeService.java

```diff
@@ -321,4 +337,37 @@ default void verifyOrder(MarketOrder marketOrder) {
   default Collection<Order> getOrder(OrderQueryParams... orderQueryParams) throws IOException {
     throw new NotAvailableFromExchangeException();
   }
+
+  /**
+   * get's the latest order form the order book that with matching orderQueryParams and populate missing data
+   *
+   * @return the order as it is on the exchange.
+   * @throws ExchangeException - Indication that the exchange reported some kind of error with the
+   *     request or response
+   * @throws NotAvailableFromExchangeException - Indication that the exchange does not support the
+   *     requested function or data
+   * @throws NotYetImplementedForExchangeException - Indication that the exchange supports the
+   *     requested function or data, but it has not yet been implemented
+   * @throws IOException - Indication that a networking error occurred while fetching JSON data
+   */
+  default Collection<Order> getOrderAssureResultsFilled(OrderQueryParams... orderQueryParams)
+      throws IOException {
+    return assureTradeResultsFilled(getOrder(orderQueryParams));
+  }
+
+  /**
+   * Populate missing data for orders.
+   *
+   * @return the order as it is on the exchange.
+   * @throws ExchangeException - Indication that the exchange reported some kind of error with the
+   *     request or response
+   * @throws NotAvailableFromExchangeException - Indication that the exchange does not support the
+   *     requested function or data
+   * @throws NotYetImplementedForExchangeException - Indication that the exchange supports the
+   *     requested function or data, but it has not yet been implemented
+   * @throws IOException - Indication that a networking error occurred while fetching JSON data
+   */
+  default Collection<Order> assureTradeResultsFilled(Collection<Order> orders) throws IOException {
```

Comment:
Only issue with this is i'm calling getOrderWithTradeResults(..) by default for multiple exchanges instead of getOrders(...) and need to pass through the orders even if nothing is done on getOrderWithTradeResults(...) for that exchange.